qiskit_experiments/calibration_management/base_calibration_experiment.py

```diff
@@ -0,0 +1,346 @@
+    and update the calibrations instance once the backend has returned the data.
+
+    Developers that wish to create a calibration experiment must subclass this base
+    class. If the experiment uses custom schedules, which is typically the case, then
+    developers may chose to use the :meth:`get_schedules` method when creating the
+    circuits for the experiment. If :meth:`get_schedules` is used then the developer
+    must override at least one of the following methods used by :meth:`get_schedules`
+    to set the schedules:
+
+    #. :meth:`get_schedules_from_options`
+
+    #. :meth:`get_schedules_from_calibrations`
+
+    #. :meth:`get_schedules_from_defaults`
+
+    These methods are called by :meth:`get_schedules`. Furthermore, developers must implement
+    the :meth:`update_calibrations` which is responsible for updating the values of the
+    parameters stored in an instance of :class:`Calibrations`. This may require the developer
+    to set the class variable :code:`__updater__` if he wishes to use the update classes
+    implemented in :mod:`qiskit_experiments.calibration_management.update_library`. In addition
+    to these calibration specific requirements, the developer must set the analysis method with
+    the class variable :code:`__analysis_class__` and any default experiment options.
+    """
+
+    # The updater class that updates the Calibrations instance
+    __updater__ = None
+
+    def __init__(self, qubits: Iterable[int], experiment_type: Optional[str] = None):
+        """Initialize the experiment object.
+
+        Args:
+            qubits: the number of qubits or list of physical qubits for
+                    the experiment.
+            experiment_type: Optional, the experiment type string.
+        """
+        super().__init__(qubits, experiment_type)
+
+        self._calibration_options = self._default_calibration_options()
+
+    def update_calibrations(self, experiment_data: ExperimentData):
+        """Update parameter values in the :class:`Calibrations` instance.
+
+        The default behaviour is to call the update method of the class variable
+        :code:`__updater__` with simplistic options. Subclasses can override this
+        method to update the instance of :class:`Calibrations` if they require a
+        more sophisticated behaviour as is the case for the :class:`Rabi` and
+        :class:`FineAmplitude` calibration experiments.
+        """
+        self.__updater__.update(
+            self.calibration_options.calibrations,
+            experiment_data,
+            parameter=self.calibration_options.cal_parameter_name,
+            schedule=self.calibration_options.schedule_name,
+        )
+
+    def get_schedule_from_options(self, option_name: str) -> ScheduleBlock:
+        """Get a schedule from the experiment options.
+
+        Developers can subclass this method if they need a more sophisticated
```

Comment:
Also adding developer docs to public API methods will look odd in the API docs for subclasses that don't override them. If it is made an internal API method this isn't an issue.